### PR TITLE
 Add the distributed source control management tool mercurial.

### PR DESCRIPTION
**Pull request by sarlalian**
- [X] I solemnly swear that this is all original content of which I am the original author
- [X] Pull request title is prepended with `[language/lang-code]`
- [X] Pull request touches only one file (or a set of logically related files with similar changes made)
- [X] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [X] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [X] Yes, I have double-checked quotes and field names!

### CODE REVIEW

**Comment by divayprakash** [comment]
@sarlalian Thank you for the contribution! Could you please correct the error in the YAML frontmatter that's causing the Travis CI build to fail?

**Comment by divayprakash** [comment]
@sarlalian is this ready to be merged?

File: mercurial.html.markdown [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+anyone working with versioned files.
+
+## Versioning Concepts
+
+### What is version control?
+
+Version control is a system that keeps track fo changes to a set of file(s)
+and/or directorie(s) over time.
+
+
```
Comment (divayprakash) [inline mercurial.html.markdown]: Please remove the double newlines here and at other places in the doc. A single empty line would suffice  for this.

File: mercurial.html.markdown [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+### Why Use Mercurial
+
+* Distributed Architecture
+* Fast
+* Platform Independent
+* Extensible
+* Easy to use
+* Open Source
+
```
Comment (divayprakash) [inline mercurial.html.markdown]: Could you merge the two lists into one? The text in the second part can just be integrated in the first list. For example -

```
* Distributed Architecture - Traditionally version control systems such as CVS and Subversion are a client server architecture with a central server to store the revsion history of a project.  Mercurial however is a truly distributed architecture, giving each devloper a full local copy of the entire development history.  It works independently of a central server.
```

Comment (divayprakash) [reply mercurial.html.markdown]: Keeping in mind line length of 80 chars. I missed that in my comment above.

File: mercurial.html.markdown [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+Mercurial is free software licensed under the terms of the [GNU General Public
+License Version 2](http://www.gnu.org/licenses/gpl-2.0.txt) or any later
+version.
+
+
+## Terminology
+
+| Term          | Definition                         |
+| ------------- | ---------------------------------- |
+| Repository    | A repository is a collection of revisions            |
```
Comment (divayprakash) [inline mercurial.html.markdown]: Remove excess spacing here as follows -

```
| Repository    | A repository is a collection of revisions |
```

**Comment by sarlalian** [comment]
I think I got what you wanted with merging the lists, let me know if I am handling them the way you want.  I can also just re-write and simplify that section.

**Comment by divayprakash** [comment]
@sarlalian Thank you for this awesome contribution! Welcome to the LXIYM community. I hope you stick around.

**Pull request merged**